### ISSUE TITLE
v1 API reference

### ISSUE THREAD

**Issue by tomwayson** [Esri/cedar#336]
Ideally we'd generate this, but apparently [typedoc + monorepo isn't so great](https://github.com/ArcGIS/arcgis-rest-js/issues/2). We can adopt @patrickarlt's [solution](https://github.com/ArcGIS/arcgis-rest-js/pull/4), which would push us towards using [acetate](http://acetate.io/) for the site, which works for me.

It's going to take a while to get all that wired up, so in the mean time, we're going to _something_ that covers at least what's in the [v0.x API page](https://esri.github.io/cedar/api/).

**Comment by ajturner** [comment]
@jgravois I'd like your input as well on what we should use for Cedar doc. Generally, what do we want to use for Hub dev docs so it's easy & consistent across all of our projects?

@akharris I'm also interested to learn what you've been exploring for Hub API doc tools

**Comment by akharris** [comment]
@ajturner I've mostly been looking for documenting web apis but I ran across a few interesting projects for documenting libraries.

### [dox](https://github.com/tj/dox) 

returns structured JSON that can be rendered using markdown. I think we could output a JSON that's compatible with our page layouts which could be reusable across the board (i.e. auto-generated docs that comes with each site, or maybe is turned on/off as a capability)

A downside is that the project is not very active (last commit is about a year ago)

## [docsify](https://github.com/QingWei-Li/docsify)

Is a good option for creating a clean doc page with linkable subsections and a nice sidebar. The docs for using docsify is a docsify site itself (I think it's deployed to GH pages).

There are a few other good ones I'll post in here

**Comment by benstoltz** [comment]
@tomwayson Do you know if there's been any changes / improvements towards a typescript doc generator + a monorepo? I know back a few months that there were issues with it, but I didn't know if there were any changes since then.

**Comment by jgravois** [comment]
i actually don't mind the TypeDoc monorepo solution that pat came up with for `Esri/arcgis-rest-js` at all and i think the page design looks great too. https://arcgis-rest-js.surge.sh

Andrew's ultimate goal (and one i agree is worth pursuing) is to be able to land developer doc like this right inside customer sites (so that **City X** can host _documentation for developers_ instead of link to [vendor documentation]()). i'm just taking a wild guess but i _assume_ 😬  you can suck up static pages created using Acetate into sub directories in ember-land.

this isn't altogether unlike deciding whether to use to use bootstrap or calcite-web. consistency is nice and we don't want to reinvent the wheel over and over again, but there's something to be said for letting maintainers choose what feels the most comfortable too (so we never lose that [sphinx flavor](http://esri.github.io/arcgis-python-api/apidoc/html/)).

tl;dr i'm not opinionated enough to be helpful.

**Comment by tomwayson** [comment]
To be clear it wasn't that I didn't _like_ Pat's solution - I think it's friggin' amazing. 🙇 

I'd love to ~~steal~~ copy that and use it for cedar. The issue is that it's not just copy and config. I have a vague idea of how it works, but suspect it would take me several hours to get that up and running over here.

To answer @benstoltz's question, I just did a cursory search, and didn't see that anything new has cropped up since I first looked for an install and config solution to using typedoc in a monorepo back in July.

TLDR: I think we should go w/ Pat's solution, I just think that it's going to take a big block of time that no one has.

**Comment by tomwayson** [comment]
re: 

> what do we want to use for Hub dev docs so it's easy & consistent across all of our projects?

well, I think most solutions for auto-docing the API are tied to the API's language. Some (like dox suggested by @akharris and typedoc) allow you to export to JSON that you can pipe into whatever static site generator (assemble, acetate, etc) you like.

@jgravois I think what @ajturner is asking (which is a fun game BTW) is can we start to converge on which static site generator we use once we've gotten the JSON for the API docs. You can imagine a plugin system for say acetate for dox JSON, typedoc JSON, etc.

**Comment by akharris** [comment]
should we schedule a video chat to discuss the issue at hand? It would be nice to have a consistent doc solution across all Hub and Hub-friendly efforts

**Comment by tomwayson** [comment]
resolved in #457
